Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathProjection.php

```diff
@@ -629,11 +629,13 @@ private function whenNodeAggregateWasMoved(NodeAggregateWasMoved $event): void
 
                 match ($newLocation->destination::class) {
                     SucceedingSiblingNodeMoveDestination::class => $this->moveNode(
+                        /** @var SucceedingSiblingNodeMoveDestination $newLocation->destination */
```

Comment:
@grebaldi i remember you used the same match class logic in the component engine but phpstan didnt complain there?

... There must be a better way than php doc... its just sad to look at @nezaniel ^^

Reply:
didn't find one, maybe we can remove it after some phpstan update

Reply:
but we use this match on classes in other places aswell? Like projections and command handler?

Reply:
see https://github.com/neos/neos-development-collection/blob/88cb1fd2a1ad3bb0e012b3594a7a16a0d7e2848a/Neos.ContentRepository.Core/Classes/Feature/NodeAggregateCommandHandler.php#L115

`/** @phpstan-ignore-next-line */` is your best friend

Reply:
I believe that if `$destination = $newLocation->destination` were established before the match clause, so that we'd actually do:

```php
$destination = $newLocation->destination;
match ($destination::class) {
    /* ... */
};
```

phpstan would be able to narrow the type of `$destination` through the match arms. I'm not sure though.

Reply:
see issue https://github.com/neos/neos-development-collection/issues/4354, we should fix this first in other places as well. So in favor of not blocking bernhards pr id like to move the discussion there ^^